File: 240131/자리 바꾸기 2/changing-seats-2.js
const [[n, k], ...inputs] = `${require("fs").readFileSync(0)}`.trim().split`
`.map(row => row.trim().split` `.map(Number))

const seats_info = Array(n + 1).fill().map((_, i) => i);
const cnt = Array(n + 1).fill().map((_, i) => new Set([i]))

Array(3).fill().forEach(_ => inputs.forEach(([a, b]) => {
    [seats_info[a], seats_info[b]] = [seats_info[b], seats_info[a]]
    cnt[seats_info[a]].add(a);
    cnt[seats_info[b]].add(b);
}))

// console.log({ cnt })

for (let i = 1; i <= n; i++) {
    console.log(cnt[i].size)
}

File: 240131/자리 바꾸기 2/changing-seats-2.py
n,k = map(int, input().split())
inputs = [list(map(int, input().split())) for _ in range(k)]
seats, cnt = [], []
for i in range(n+1):
    seats.append(i)
    cnt.append(set([i]))

for _ in range(3):
    for a,b in inputs:
        seats[a],seats[b] = seats[b], seats[a]
        cnt[seats[a]].add(a)
        cnt[seats[b]].add(b)

for i in range(1,n+1):
    print(len(cnt[i]))